From 14ebd8879a901e791173f92193a2645eee2b0c16 Mon Sep 17 00:00:00 2001
From: Srujana Challa <schalla@marvell.com>
Date: Thu, 28 Jan 2021 15:02:14 +0530
Subject: [PATCH 1233/1921] octeontx2-af: Fix setting of CPT_AF_X2PX_LINK_CFG

Fix setting of CPT_AF_X2PX_LINK_CFG register with
incorrect value.

Fixes: 'commit 21d1b6c2da8b ("octeontx2-af: config for cn10k
inbound inline ipsec")'

Signed-off-by: Srujana Challa <schalla@marvell.com>
Change-Id: I8d905eab754aee8d9f55e3b39c2dc32e6bd8792f
Reviewed-on: https://sj1git1.cavium.com/c/IP/SW/kernel/linux/+/44916
Tested-by: sa_ip-sw-jenkins <sa_ip-sw-jenkins@marvell.com>
Reviewed-by: Vidya Sagar Velumuri <vvelumuri@marvell.com>
Reviewed-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
[WK: The original patch got from Marvell sdk11.21.09]
Signed-off-by: Wenlin Kang <wenlin.kang@windriver.com>
---
 drivers/net/ethernet/marvell/octeontx2/af/rvu_cpt.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/net/ethernet/marvell/octeontx2/af/rvu_cpt.c b/drivers/net/ethernet/marvell/octeontx2/af/rvu_cpt.c
index a893e40f6e39..9095df3ae5a3 100644
--- a/drivers/net/ethernet/marvell/octeontx2/af/rvu_cpt.c
+++ b/drivers/net/ethernet/marvell/octeontx2/af/rvu_cpt.c
@@ -380,7 +380,7 @@ static int cpt_inline_ipsec_cfg_inbound(struct rvu *rvu, int blkaddr, u8 cptlf,
 	 * range of channels it should propagate to X2P
 	 */
 	if (!is_rvu_otx2(rvu)) {
-		val |= (ilog2(NIX_CHAN_CPT_X2P_MASK + 1) << 16);
+		val = (ilog2(NIX_CHAN_CPT_X2P_MASK + 1) << 16);
 		val |= rvu->hw->cpt_chan_base;
 
 		rvu_write64(rvu, blkaddr, CPT_AF_X2PX_LINK_CFG(0), val);
-- 
2.31.1

